### ISSUE TITLE
repair.go:73 Operation cannot be fulfilled on servicenodeportallocations "": another caller has already initialized the resource

### ISSUE THREAD

**Issue by jhughes2112** [rancher/rancher#24787]
**What kind of request is this:** bug

**Steps to reproduce:** Boot two or more identical diskless machines (I have bare metal, but VM might suffice) and PXE boot them to a RancherOS image.  Simultaneously in multiple SSH windows connected to the nodes, run the rancher-agent docker command provided at the bottom of a new Cluster description.  Make sure etcd, control plane, and worker are all enabled.

**Result:** It is not difficult to have one or more machines be allocated the same IP address for one of the services.  This shows up as a cluster that hangs on provisioning because one node is not responding to kube-api calls due to invalid IP address.

 (I'm not an expert, trying to troubleshoot my first cluster.)

Here's a good docker log from kube-apiserver on a working node (10.0.0.245):
```
I1227 20:45:00.044803       1 secure_serving.go:123] Serving securely on [::]:6443
I1227 20:45:00.044892       1 apiservice_controller.go:94] Starting APIServiceRegistrationController
I1227 20:45:00.044959       1 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I1227 20:45:00.044972       1 autoregister_controller.go:140] Starting autoregister controller
I1227 20:45:00.045021       1 cache.go:32] Waiting for caches to sync for autoregister controller
I1227 20:45:00.045097       1 controller.go:81] Starting OpenAPI AggregationController
I1227 20:45:00.045143       1 available_controller.go:383] Starting AvailableConditionController
I1227 20:45:00.045166       1 crd_finalizer.go:274] Starting CRDFinalizer
I1227 20:45:00.045194       1 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I1227 20:45:00.045278       1 crdregistration_controller.go:111] Starting crd-autoregister controller
I1227 20:45:00.045303       1 controller.go:85] Starting OpenAPI controller
I1227 20:45:00.045324       1 establishing_controller.go:73] Starting EstablishingController
I1227 20:45:00.045374       1 nonstructuralschema_controller.go:191] Starting NonStructuralSchemaConditionController
I1227 20:45:00.045423       1 customresource_discovery_controller.go:208] Starting DiscoveryController
I1227 20:45:00.045444       1 apiapproval_controller.go:185] Starting KubernetesAPIApprovalPolicyConformantConditionController
I1227 20:45:00.045483       1 naming_controller.go:288] Starting NamingConditionController
I1227 20:45:00.045309       1 shared_informer.go:197] Waiting for caches to sync for crd-autoregister
E1227 20:45:00.047334       1 controller.go:154] Unable to remove old endpoints from kubernetes service: StorageError: key not found, Code: 1, Key: /registry/masterleases/10.0.0.245, ResourceVersion: 0, AdditionalErrorMsg:
I1227 20:45:00.145184       1 cache.go:39] Caches are synced for APIServiceRegistrationController controller
I1227 20:45:00.145254       1 cache.go:39] Caches are synced for autoregister controller
I1227 20:45:00.145440       1 cache.go:39] Caches are synced for AvailableConditionController controller
I1227 20:45:00.145646       1 shared_informer.go:204] Caches are synced for crd-autoregister
F1227 20:45:00.425434       1 controller.go:163] Unable to perform initial IP allocation check: Operation cannot be fulfilled on serviceipallocations "": another caller has already initialized the resource
```

And here's where the broken one (10.0.0.243) began deviating (notice the time is within one second of the previous machine, which makes me believe it is a race condition, possibly):
```
I1227 20:44:59.851877       1 secure_serving.go:123] Serving securely on [::]:6443
I1227 20:44:59.851972       1 controller.go:81] Starting OpenAPI AggregationController
I1227 20:44:59.852134       1 crd_finalizer.go:274] Starting CRDFinalizer
I1227 20:44:59.852317       1 available_controller.go:383] Starting AvailableConditionController
I1227 20:44:59.852413       1 cache.go:32] Waiting for caches to sync for AvailableConditionController controller
I1227 20:44:59.852433       1 establishing_controller.go:73] Starting EstablishingController
I1227 20:44:59.852432       1 apiservice_controller.go:94] Starting APIServiceRegistrationController
I1227 20:44:59.852497       1 nonstructuralschema_controller.go:191] Starting NonStructuralSchemaConditionController
I1227 20:44:59.852503       1 cache.go:32] Waiting for caches to sync for APIServiceRegistrationController controller
I1227 20:44:59.852469       1 apiapproval_controller.go:185] Starting KubernetesAPIApprovalPolicyConformantConditionController
I1227 20:44:59.852576       1 crdregistration_controller.go:111] Starting crd-autoregister controller
I1227 20:44:59.852611       1 shared_informer.go:197] Waiting for caches to sync for crd-autoregister
I1227 20:44:59.852711       1 controller.go:85] Starting OpenAPI controller
I1227 20:44:59.852668       1 naming_controller.go:288] Starting NamingConditionController
I1227 20:44:59.852861       1 customresource_discovery_controller.go:208] Starting DiscoveryController
I1227 20:44:59.852136       1 autoregister_controller.go:140] Starting autoregister controller
I1227 20:44:59.853133       1 cache.go:32] Waiting for caches to sync for autoregister controller
E1227 20:44:59.855915       1 controller.go:154] Unable to remove old endpoints from kubernetes service: StorageError: key not found, Code: 1, Key: /registry/masterleases/10.0.0.243, ResourceVersion: 0, AdditionalErrorMsg:
I1227 20:44:59.952683       1 cache.go:39] Caches are synced for AvailableConditionController controller
I1227 20:44:59.952794       1 cache.go:39] Caches are synced for APIServiceRegistrationController controller
I1227 20:44:59.952812       1 shared_informer.go:204] Caches are synced for crd-autoregister
I1227 20:44:59.953390       1 cache.go:39] Caches are synced for autoregister controller
E1227 20:45:00.257467       1 repair.go:73] Operation cannot be fulfilled on servicenodeportallocations "": another caller has already initialized the resource
E1227 20:45:00.267854       1 repair.go:100] Operation cannot be fulfilled on serviceipallocations "": another caller has already initialized the resource
I1227 20:45:00.852125       1 controller.go:107] OpenAPI AggregationController: Processing item
I1227 20:45:00.852169       1 controller.go:130] OpenAPI AggregationController: action for item : Nothing (removed from the queue).
I1227 20:45:00.852197       1 controller.go:130] OpenAPI AggregationController: action for item k8s_internal_local_delegation_chain_0000000000: Nothing (removed from the queue).
I1227 20:45:00.862704       1 storage_scheduling.go:139] created PriorityClass system-node-critical with value 2000001000
I1227 20:45:00.867113       1 storage_scheduling.go:139] created PriorityClass system-cluster-critical with value 2000000000
I1227 20:45:00.867130       1 storage_scheduling.go:148] all system priority classes are created successfully or already exist.
I1227 20:45:01.394922       1 controller.go:606] quota admission added evaluator for: roles.rbac.authorization.k8s.io
I1227 20:45:01.495103       1 controller.go:606] quota admission added evaluator for: rolebindings.rbac.authorization.k8s.io
I1227 20:47:15.862229       1 log.go:172] http: TLS handshake error from 10.0.0.244:34609: EOF
```

Error message in Rancher UI during provisioning shows 10.0.0.245 having a problem with IP allocation, whereas 10.0.0.243 was complaining about similar issues with IP allocation:
```
[controlPlane] Failed to bring up Control Plane: [Failed to verify healthcheck: Service [kube-apiserver] is not healthy on host [10.0.0.245]. Response code: [403], response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"forbidden: User \"kube-apiserver\" cannot get path \"/healthz\": RBAC: clusterrole.rbac.authorization.k8s.io \"system:public-info-viewer\" not found","reason":"Forbidden","details":{},"code":403} , log: E1228 03:35:43.895931 1 controller.go:224] unable to sync kubernetes service: Service "kubernetes" is invalid: spec.clusterIP: Invalid value: "10.43.0.1": provided IP is already allocated]
```

**Environment information**
- Rancher version : 2.3.2
- Installation option: single Rancher server docker image 

**Cluster information**
- Cluster type: Custom/none
- Machine type and specifications: Bare metal, diskless, pxe booting RancherOS with static IPs
- Kubernetes version (use `kubectl version`): v1.16.3-rancher1-1
- Docker version: 18.09

I'm going to try booting each machine in sequence and see if this helps.  Unfortunately, anytime a machine fails to provision, I have to wipe the entire configuration of the Rancher Server and start fresh, otherwise nodes fail differently.

**Comment by hongfeng-yaya** [comment]
I also encountered the same problem, have you solved it?

**Comment by jhughes2112** [comment]
No, I gave up on PXE booting, specifically, because RancherOS would run out of file handles or ram disk space.  I did believe there was a race condition when multiple machines were starting at the same time, though, which was this particular bug.  Given that I was PXE booting a diskless machine, this could not be a "dirty" machine that needed cleaning, so it was a pretty good indication something was wrong.
